arcgis-ios-sdk-samples/Maps/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningViewController.swift

```diff
@@ -0,0 +1,312 @@
+                    featureLayer.definitionExpression = "VERTICAL_ORDER = \(floor)"
+                }
+            }
+        }
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["ShowDeviceLocationUsingIndoorPositioningViewController"]
+    }
+    
+    deinit {
+        // Stop location display, which in turn stop the data source.
+        mapView.locationDisplay.stop()
+        // Reset indoors location data source.
+        indoorsLocationDataSource?.locationChangeHandlerDelegate = nil
+        indoorsLocationDataSource = nil
+        // Reset the API key.
+        AGSArcGISRuntimeEnvironment.apiKey = apiKey
+    }
+}
+
+// MARK: - AGSLocationChangeHandlerDelegate
+
+extension ShowDeviceLocationUsingIndoorPositioningViewController: AGSLocationChangeHandlerDelegate {
+    /// The delegate method to receive location updates from the data source.
+    /// - Parameters:
+    ///   - locationDataSource: The indoors location data source.
+    ///   - location: The indoors location.
+    func locationDataSource(_ locationDataSource: AGSLocationDataSource, locationDidChange location: AGSLocation) {
+        // The floor level provided by the indoors beacons.
+        let floorText: String
+        if let floor = location.additionalSourceProperties[.floor] as? Int {
+            currentFloor = floor
+            floorText = String(format: "Floor level: %d", floor)
+        } else {
+            floorText = "Floor not available"
+        }
+        
+        // The horizontal accuracy of the positioning signal from the sensors.
+        let horizontalAccuracy = measurementFormatter.string(from: Measurement(value: Double(location.horizontalAccuracy), unit: UnitLength.meters))
+        
+        // Possible sources: GNSS, AppleIPS, BLE, WIFI, CELL, IP.
+        let positionSource = location.additionalSourceProperties[.positionSource] as? String ?? "NA"
+        let transmitterCount = location.additionalSourceProperties[AGSLocationSourcePropertyKey(rawValue: "transmitterCount")] as? Int ?? 0
+        let satelliteCount = location.additionalSourceProperties[.satelliteCount] as? Int ?? 0
+        let sensorCount: String = {
+            switch positionSource {
+            case "GNSS":
+                return String(format: "%d satellite(s)", satelliteCount)
+            default:
+                return String(format: "%d beacon(s)", transmitterCount)
+            }
+        }()
+        
+        DispatchQueue.main.async { [weak self] in
+            guard let self = self else { return }
+            self.sourceStatusLabel.text = String(format: "%@, Position source: %@", floorText, positionSource)
+            self.sensorStatusLabel.text = String(format: "%@, Horizontal accuracy: %@", sensorCount, horizontalAccuracy)
```

Comment:
I'm testing with a device not in building L. But the horizontal accuracy has a 1.219m value. Is this correct? In cases where the user isn't in the building, should the horizontal accuracy be 0 or N/A?

Reply:
The accuracy will be from the other location data provider, in your case could be GPS.
I don't know why it is so accurate though - should be 1~10m IMO, but might be just how our API works. 😅